pkg/apis/core/types.go

```diff
@@ -2662,6 +2662,10 @@ type PodSpec struct {
 	// The higher the value, the higher the priority.
 	// +optional
 	Priority *int32
+	// Preempting specifies whether a pod with this PriorityClass could start a preemption process.
+	// If not specified, the default is true.
+	// +optional
+	Preempting *bool
```

Comment:
per the [API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md), many boolean fields tend to grow to multi-value enums in the future:

> Think twice about bool fields. Many ideas start as boolean but eventually trend towards a small set of mutually exclusive options. Plan for future expansions by describing the policy options explicitly as a string type alias (e.g. TerminationMessagePolicy).

a `preemptionPolicy` field with a default value of `PreemptLowerPriority`, and a newly-allowed value of `Never` might be easier to understand (names are strawmen, feedback is welcome)

we typically do enums as typed strings (see `ImagePullPolicy PullPolicy` as an example), and in this case it would be a pointer, since the field is optional

we would need accompanying validation to make sure the specified policy enum value was valid/recognized.

Reply:
since this is adding a new field to a stable API, it has to start as alpha, and not allow new data into the field for one release (unless the alpha feature gate is enabled).

see https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#adding-unstable-features-to-stable-versions for the process of doing this

add to the docs for every field that the field is alpha level and which feature gate controls its enablement

Reply:
Thanks, Jordan @liggitt  I add the new field fellow the process. And I think there is just tow choice for the preemptionPolicy: true of false? Do we need to change boolean fields to multi-value enums? @bsalamat  WDYT?

Reply:
@denkensk Yes, please change it to multi-value enums. For now the possible values will be `Never` and `PreemptLowerPriority`. Default value will be `PreemptLowerPriority`. In the future, we might add more choices.

Reply:
Actually, you can drop `Preempt` from `PreemptLowerPriority` and use `LowerPriority` only.